### DISCUSSION TITLE
Which notification types currently work in `4.1.0`?

### DISCUSSION THREAD

**Discussion by alec-pinson** [theyosh/TerrariumPI#648] [Q&A] [answered]
I get an error when trying to use pushover but I read some stuff about you're moving to webhook, messaging and scripts only?

including the error for more info
```
terrariumpi_1  | Exception in thread Thread-1466:
terrariumpi_1  | Traceback (most recent call last):
terrariumpi_1  |   File "/usr/local/lib/python3.8/threading.py", line 932, in _bootstrap_inner
terrariumpi_1  |     self.run()
terrariumpi_1  |   File "/usr/local/lib/python3.8/threading.py", line 870, in run
terrariumpi_1  |     self._target(*self._args, **self._kwargs)
terrariumpi_1  |   File "/TerrariumPI/terrariumEngine.py", line 575, in _update_sensors
terrariumpi_1  |     self.notification.message(f'sensor_update' , sensor_data)
terrariumpi_1  |   File "/TerrariumPI/terrariumNotification.py", line 328, in message
terrariumpi_1  |     if self.services[service.id] is None:
terrariumpi_1  | KeyError: 'a1a2345a-a123-1a23-12a3-12a6aaa345a'
```

**Comment by alec-pinson** [comment, marked as answer]
weirdly rebooted my pi and the errors gone.

**Reply by theyosh** [reply]
That is always funny ;)

**Comment by theyosh** [comment]
The notification system is almost finished. Just make sure you have the latest code of the 4.x.y.z branch if you need notification. The main branch is a bit behind. The next release 4.2.0 should have a fully working notification system.

Pushover should work. I use the webhook method, because I only need to broadcast to Pusher, and not receiving. So that should not be an issue. I have had my pusher spamming with messages and sounds :)

But, just to make sure, use branch 4.x.y.z for testing and make sure you have the latest code.

**Reply by alec-pinson** [reply]
Yea i originally tried pushover and also webhook but both were getting the above error, rebooted my pi today and started getting notifications come through on pushover lol.

I'm currently using the latest docker image 😄
